safeweb/http.go

```diff
@@ -203,7 +207,9 @@ func NewServer(config Config) (*Server, error) {
 	if config.CSPAllowInlineStyles {
 		s.csp = defaultCSP + `; style-src 'self' 'unsafe-inline'`
 	}
-	s.h = &http.Server{Handler: s}
+
+	s.h = cmp.Or(config.HTTPServer, &http.Server{})
```

Comment:
optional:
```suggestion
	s.h = cmp.Or(config.HTTPServer, &http.Server{})
	if s.h.Handler != nil {
		return nil, fmt.Errorf("use safeweb.Config.APIMux and safeweb.Config.BrowserMux instead of http.Server.Handler")
	}
```